frontend/src/hooks/api/pkiCollections/mutations.tsx

```diff
@@ -0,0 +1,97 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { apiRequest } from "@app/config/request";
+
+import { workspaceKeys } from "../workspace/queries";
+import { pkiCollectionKeys } from "./queries";
+import {
+  TAddItemToPkiCollectionDTO,
+  TCreatePkiCollectionDTO,
+  TDeletePkiCollectionDTO,
+  TPkiCollection,
+  TPkiCollectionItem,
+  TRemoveItemFromPkiCollectionDTO,
+  TUpdatePkiCollectionTO} from "./types";
+
+export const useCreatePkiCollection = () => {
+  const queryClient = useQueryClient();
+  return useMutation<TPkiCollection, {}, TCreatePkiCollectionDTO>({
+    mutationFn: async (body) => {
+      const { data: pkiCollection } = await apiRequest.post<TPkiCollection>(
+        "/api/v1/pki/collections",
+        body
+      );
+      return pkiCollection;
+    },
+    onSuccess: (_, { projectId }) => {
+      queryClient.invalidateQueries(workspaceKeys.getWorkspacePkiCollections(projectId));
+    }
+  });
+};
+
+export const useUpdatePkiCollection = () => {
+  const queryClient = useQueryClient();
+  return useMutation<TPkiCollection, {}, TUpdatePkiCollectionTO>({
+    mutationFn: async ({ collectionId, ...body }) => {
+      const { data: pkiCollection } = await apiRequest.patch<TPkiCollection>(
+        `/api/v1/pki/collections/${collectionId}`,
+        body
+      );
+      return pkiCollection;
+    },
+    onSuccess: (_, { projectId, collectionId }) => {
+      queryClient.invalidateQueries(workspaceKeys.getWorkspacePkiCollections(projectId));
+      queryClient.invalidateQueries(pkiCollectionKeys.getPkiCollectionById(collectionId));
+    }
+  });
+};
+
+export const useDeletePkiCollection = () => {
+  const queryClient = useQueryClient();
+  return useMutation<TPkiCollection, {}, TDeletePkiCollectionDTO>({
+    mutationFn: async ({ collectionId }) => {
+      const { data: pkiCollection } = await apiRequest.delete<TPkiCollection>(
+        `/api/v1/pki/collections/${collectionId}`
+      );
+      return pkiCollection;
+    },
+    onSuccess: (_, { projectId, collectionId }) => {
+      queryClient.invalidateQueries(workspaceKeys.getWorkspacePkiCollections(projectId));
+      queryClient.invalidateQueries(pkiCollectionKeys.getPkiCollectionById(collectionId));
+    }
+  });
+};
+
+export const useAddItemToPkiCollection = () => {
+  const queryClient = useQueryClient();
+  return useMutation<TPkiCollectionItem, {}, TAddItemToPkiCollectionDTO>({
+    mutationFn: async ({ collectionId, type, itemId }) => {
+      const { data: pkiCollectionItem } = await apiRequest.post<TPkiCollectionItem>(
+        `/api/v1/pki/collections/${collectionId}/items`,
+        {
+          type,
+          itemId
+        }
+      );
+      return pkiCollectionItem;
+    },
+    onSuccess: (_, { collectionId }) => {
+      queryClient.invalidateQueries(pkiCollectionKeys.getPkiCollectionItems(collectionId));
+    }
+  });
+};
+
+export const useRemoveItemFromPkiCollection = () => {
```

Comment:
I think an X icon would be better. The thing with the "trash" icon is that it misleads me to think that the cert/ca will actually get deleted. What do you think? 
<img width="913" alt="image" src="https://github.com/user-attachments/assets/59429ac2-4e71-4a61-a7df-2c58969b1ad9">

Reply:
Adjusted!